shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -329,6 +332,46 @@ - (void)loadView {
   self.view.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
 
   [self installSplashScreenViewIfNecessary];
+  // This is a workaround to accomodate iOS 13 and higher.  There isn't a way to get touches on the
+  // status bar to trigger scrolling to the top of a scroll view.  We place a UIScrollView offscreen
+  // with a content offset so we can get those events. See also:
+  // https://github.com/flutter/flutter/issues/35050
+  UIScrollView* scrollView = [[UIScrollView alloc] init];
+  scrollView.autoresizingMask = UIViewAutoresizingFlexibleWidth;
+  // The color shouldn't matter since it is offscreen.
+  scrollView.backgroundColor = UIColor.whiteColor;
+  scrollView.delegate = self;
+  // This is an arbitrary small size.
+  scrollView.contentSize = CGSizeMake(10, 10);
+  // This is an arbitrary offset that is not CGPointZero.
+  scrollView.contentOffset = CGPointMake(10, 10);
+  // This is some aribrary place offscreen.
+  scrollView.bounds = CGRectMake(0, -10, 0, 10);
+  [self.view addSubview:scrollView];
+  _scrollView.reset(scrollView);
+}
+
+static void sendFakeTouchEvent(FlutterEngine* engine,
+                               CGPoint location,
+                               flutter::PointerData::Change change) {
+  const CGFloat scale = [UIScreen mainScreen].scale;
+  flutter::PointerData pointer_data;
+  pointer_data.Clear();
+  pointer_data.physical_x = location.x * scale;
+  pointer_data.physical_y = location.y * scale;
+  pointer_data.kind = flutter::PointerData::DeviceKind::kTouch;
+  pointer_data.time_stamp = [[NSDate date] timeIntervalSince1970] * kMicrosecondsPerSecond;
+  auto packet = std::make_unique<flutter::PointerDataPacket>(/*count=*/1);
+  pointer_data.change = change;
+  packet->SetPointerData(0, pointer_data);
+  [engine dispatchPointerDataPacket:std::move(packet)];
+}
+
+- (BOOL)scrollViewShouldScrollToTop:(UIScrollView*)scrollView {
+  CGPoint statusBarPoint = CGPointZero;
+  sendFakeTouchEvent(_engine.get(), statusBarPoint, flutter::PointerData::Change::kDown);
```

Comment:
Wouldnt it make more sense to send a message to flutter instead of sending fake touch events? @xster

Reply:
FWIW I did it this way since thats how the code was previously working.  I tried to change as little as possible.

Reply:
I asked heaving in mind that flutter/flutter#42560 will be implemented somewhere in the future.
But it might make more sense to change the way the tap is handled on the dart side.
I believe currently it is done by the scaffold and should probably be moved to the routes.

@xter should I make a separate issue for this or add a comment to the one linked?